core/array.rb

```diff
@@ -1516,30 +1516,13 @@ def sample(count=undefined, options=undefined)
         while i < count
           k = rng.rand(size)
 
-          spin = false
-          spin_count = 0
-
-          while true
-            j = 0
-            while j < i
-              if k == result[j]
-                spin = true
-                break
-              end
-
-              j += 1
-            end
-
-            if spin
```

Comment:
If you add spin=false here(original algorithm with no bug?), the calculation cost might be very large.